bothan-api/server-cli/src/commands/query.rs

```diff
@@ -0,0 +1,466 @@
+        .into_iter()
+        .collect::<Result<Vec<Vec<AssetInfo>>, anyhow::Error>>()?
+        .into_iter()
+        .flatten()
+        .collect::<Vec<AssetInfo>>();
+
+    print_asset_infos(asset_infos);
+
+    Ok(())
+}
+
+async fn query_bitfinex<T: Into<Duration>>(
+    opts: bothan_bitfinex::WorkerOpts,
+    query_ids: &[String],
+    timeout_interval: T,
+) -> anyhow::Result<()> {
+    let api = bothan_bitfinex::api::builder::RestApiBuilder::new(opts.url).build()?;
+
+    let distinct_ids = get_distinct_ids(query_ids);
+    let asset_infos = timeout(
+        timeout_interval.into(),
+        RestAssetInfoProvider::get_asset_info(&api, &distinct_ids),
+    )
+    .await??;
+
+    print_asset_infos(asset_infos);
+    Ok(())
+}
+
+async fn query_bybit<T: Into<Duration>>(
+    opts: bothan_bybit::WorkerOpts,
+    query_ids: &[String],
+    timeout: T,
+) -> anyhow::Result<()> {
+    let connector = bothan_bybit::WebSocketConnector::new(opts.url);
+    let mut provider = connector.connect().await?;
+
+    let distinct_ids = get_distinct_ids(query_ids);
+    WebSocketAssetInfoProvider::subscribe(&mut provider, &distinct_ids).await?;
+    let asset_infos = query(&mut provider, distinct_ids, timeout.into()).await?;
+
+    print_asset_infos(asset_infos);
+    Ok(())
+}
+
+async fn query_coinbase<T: Into<Duration> + Clone>(
+    opts: bothan_coinbase::WorkerOpts,
+    query_ids: &[String],
+    timeout: T,
+) -> anyhow::Result<()> {
+    let connector = bothan_coinbase::WebSocketConnector::new(opts.url);
+
+    let mut tasks = FuturesOrdered::new();
+
+    let distinct_ids = get_distinct_ids(query_ids);
+
+    for chunk in distinct_ids
+        .into_iter()
+        .chunks(opts.max_subscription_per_connection)
+        .into_iter()
+    {
+        let cloned_timeout = timeout.clone().into();
+        let chunk_ids = chunk.collect::<Vec<String>>();
+        let mut provider = connector.connect().await?;
+        WebSocketAssetInfoProvider::subscribe(&mut provider, &chunk_ids).await?;
+
+        tasks.push_back(async move { query(&mut provider, chunk_ids, cloned_timeout).await });
+    }
+
+    let results = tasks
+        .collect::<Vec<Result<Vec<AssetInfo>, anyhow::Error>>>()
+        .await;
+
+    let asset_infos = results
+        .into_iter()
+        .collect::<Result<Vec<Vec<AssetInfo>>, anyhow::Error>>()?
+        .into_iter()
+        .flatten()
+        .collect::<Vec<AssetInfo>>();
+
+    print_asset_infos(asset_infos);
+
+    Ok(())
+}
+
+async fn query_coingecko<T: Into<Duration>>(
+    opts: bothan_coingecko::WorkerOpts,
+    query_ids: &[String],
+    timeout_interval: T,
+) -> anyhow::Result<()> {
+    let api = bothan_coingecko::api::RestApiBuilder::new(opts.url, opts.user_agent, opts.api_key)
+        .build()?;
+
+    let distinct_ids = get_distinct_ids(query_ids);
+    let asset_infos = timeout(
+        timeout_interval.into(),
+        RestAssetInfoProvider::get_asset_info(&api, &distinct_ids),
+    )
+    .await??;
```

Comment:
a lot of duplicate code,  common_rest_api + common_websocket. Can resolve if unnecessary